### ISSUE TITLE
Bug: In the Deep Learning model, optimizers aren't completely copied to the nodes

### ISSUE THREAD

**Issue by xehartnort** [sherpaai/Sherpa.ai-Federated-Learning-Framework#165]
When a Deep Learning model is created, the keras optimizers are not completely copied in the nodes, parameters such as learning rate are set to default.
This bug can be found in the file [shfl/model/deep_learning_model.py](https://github.com/sherpaai/Sherpa.ai-Federated-Learning-Framework/blob/master/shfl/model/deep_learning_model.py), in the method `__deepcopy__` lines 130-131:
```python
130     model.compile(optimizer=v.optimizer.__class__.__name__, loss=v.loss, 
131           metrics=v.compiled_metrics._user_metrics)
```
